### PR TITLE
Cloud selection screen

### PR DESCRIPTION
**Pull request by jdk2588**
`go run main.go gcp --cf="./gcp-config.json"`

Screenshot 2023-10-23 at 5 55 10 PM

### CODE REVIEW

File: internal/view/app.go [hunk 1]
```diff
@@ -63,38 +137,79 @@ func (a *App) Init(ctx context.Context, profiles, regions []string, version stri
...
+		switch seletedCloud {
+		case "AWS":
+			a.handleAWS()
+			if err := a.command.defaultCmd(); err != nil {
+				panic(err)
+			}
+			a.Main.SwitchToPage(internal.AWS_SCREEN)
+		case "GCP":
+			a.handleGCP()
+			if err := a.command.defaultCmd(); err != nil {
```
Comment (RamanaReddy0M) [inline internal/view/app.go:170]: Seems like in any case need to call `defaultCmd`.  I think we can extract this common part out to switch block, what do you think?

File: internal/view/app.go [hunk 2]
```diff
@@ -63,38 +137,79 @@ func (a *App) Init(ctx context.Context, profiles, regions []string, version stri
...
-	a.Content.Stack.AddListener(a.Crumbs())
-	a.App.Init()
-	a.SetInputCapture(a.keyboard)
-	a.bindKeys()
+}
+
+func (a *App) handleGCP() {
+	ctx := a.GetContext()
+	ctx = context.WithValue(ctx, internal.KeySelectedCloud, "GCP")
+	ctx = context.WithValue(ctx, internal.KeyActiveProject, "centering-aegis-400910")
```
Comment (RamanaReddy0M) [inline internal/view/app.go:147]: I think we can move these context values to some const values.

**Review by RamanaReddy0M — COMMENTED** [review]
@PratikJethe can add a small description of this pull request.
In description, drop a command to run gcp.
Also add a screenshot of that command, So that we can compare how it is supposed to look.

File: internal/gcp/projects.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	"github.com/rs/zerolog/log"
+	"google.golang.org/api/iterator"
+
+	resourcemanager "cloud.google.com/go/resourcemanager/apiv3"
+	"cloud.google.com/go/resourcemanager/apiv3/resourcemanagerpb"
+)
+
+
+func  FetchProjects() {
```
[the remark this answers is not in the document]
Comment (PratikJethe) [reply internal/gcp/projects.go:14]: Yes, initially we planned to fetch the project ID through APIs but that was causing permission errors. so we decided to extract the project ID from the credential file itself. This method will be replaced with the new one

**Review by RamanaReddy0M — APPROVED** [review]

**Pull request merged**